prov/verbs/src/fi_verbs.c

```diff
@@ -357,12 +363,14 @@ ssize_t fi_ibv_send_iov_flags(struct fi_ibv_msg_ep *ep, struct ibv_send_wr *wr,
 	else
 		fi_ibv_set_sge_iov(wr->sg_list, iov, count, desc, len);
 
-	wr->send_flags = VERBS_INJECT_FLAGS(ep, len, flags) | VERBS_COMP_FLAGS(ep, flags);
+	wr->num_sge = count;
+	wr->send_flags |= VERBS_INJECT_FLAGS(ep, len, flags);
+	wr->send_flags |= VERBS_COMP_FLAGS(ep, flags);
 
 	if (flags & FI_FENCE)
-		wr->send_flags = IBV_SEND_FENCE;
+		wr->send_flags |= IBV_SEND_FENCE;
```

Comment:
This bug fix still needs to go into a separate patch.

Reply:
Done in new version.